### ISSUE TITLE
Support for sentence sampling based on probability

### ISSUE THREAD

**Issue by GokulNC** [marian-nmt/marian-dev#802]
I see that we have support for [`--data-weighting`](https://github.com/marian-nmt/marian-dev/issues/171) in Marian-NMT for sentence-level weightage to multiply with each sample's loss function.

Is there anyway we can use the sentence-weighting as a sampling probability instead of weighting factor that gets multiplied with the cost?

Use-case: I have a corpus with 10M pure translated data and 400M back-translated data.
I do not want each epoch to train on all sentences from back-translated data.

Basically, I need to do under-sampling of a specific set of data.
So is it somehow possible to set probability of sampling a sentence for the training corpus?

Thanks!
